src/ScVmm/ScVmm.Autorest/custom/Update-AzScVmmVM.ps1

```diff
@@ -0,0 +1,373 @@
+    [Alias('VMName')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Path')]
+    [System.String]
+    # The name of the virtual machine.
+    ${Name},
+
+    [Parameter(Mandatory)]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Path')]
+    [System.String]
+    # The name of the resource group.
+    # The name is case insensitive.
+    ${ResourceGroupName},
+
+    [Parameter()]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Runtime.DefaultInfo(Script = '(Get-AzContext).Subscription.Id')]
+    [System.String]
+    # The ID of the target subscription.
+    # The value must be an UUID.
+    ${SubscriptionId},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.Int32]
+    # Gets or sets the number of vCPUs for the vm.
+    ${CpuCount},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [System.Management.Automation.SwitchParameter]
+    # Whether to enable dynamic memory or not.
+    ${DynamicMemoryEnabled},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.Int32]
+    # Gets or sets the max dynamic memory for the vm.
+    ${DynamicMemoryMaxMb},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.Int32]
+    # Gets or sets the min dynamic memory for the vm.
+    ${DynamicMemoryMinMb},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [System.Management.Automation.SwitchParameter]
+    # Whether to enable processor compatibility mode for live migration of VMs.
+    ${LimitCpuForMigration},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.Int32]
+    # MemoryMB is the size of a virtual machine's memory, in MB.
+    ${MemoryMb},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.String]
+    # Type of checkpoint supported for the vm.
+    ${CheckpointType},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [AllowEmptyCollection()]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.String[]]
+    # Availability Sets in vm.
+    ${AvailabilitySetName},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [AllowEmptyCollection()]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.String[]]
+    # Availability Sets in vm.
+    ${AvailabilitySetId},
+
+    [Parameter(ParameterSetName='UpdateExpanded')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Runtime.Info(PossibleTypes=([Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Models.IResourceUpdateTags]))]
+    [System.Collections.Hashtable]
+    # Resource tags
+    ${Tags},
```

Comment:
Parameter [-Tags] of cmdlet Update-AzScVmmVM does not follow the enforced naming convention of using a singular noun for a parameter name. Please rename it to [-Tag]